src/lib/watt-time/README.md

```diff
@@ -47,81 +34,136 @@ WATT_TIME_PASSWORD: <your-password>
 
-- `timestamp`: Timestamp of the recorded event (2021-01-01T00:00:00Z) RFC3339
-- `duration`: Duration of the recorded event in seconds (3600)
-- `geolocation`: Location of the software system (latitude in decimal degrees, longitude in decimal degrees). "latitude,longitude"
-- `cloud/region-geolocation`: The same as `geolocation`, with calculations performed by the `cloud-metadata` plugin
-- `cloud/region-wt-id`: Region abbreviation associated with location (e.g. 'CAISO_NORTH')
+   - Retrieve WattTime data for the specified duration. The WattTime API adds aggregated emissions for every 5 minutes. To address this limitation, the plugin sets the previous emission's value if the specified `duration` is less than 5 minutes.
 
-Either `geolocation`,`cloud/region-wt-id` or `cloud/region-geolocation` should be provided.
+   - Calculate average emissions based on retrieved data. The WattTime API returns full data for the entire duration; the plugin checks if the data's period time is within the specified input range and collects data in `kgMWh`.
 
-### Typescript Usage
+3. **Output**: Return results with the average grid emissions for each input.
 
-```typescript
-// environment variable configuration
-// export WATT_TIME_USERNAME=test1
-// export WATT_TIME_PASSWORD=test2
-// use environment variables to configure the plugin
-const output = WattTimeGridEmissions();
-const result = await output.execute([
+### TypeScript Usage
+
+```ts
+// Initialize the plugin
+const plugin = WattTimeGridEmissions();
+
+// Execute the plugin with input parameters
+const inputs = [
   {
-    timestamp: '2021-01-01T00:00:00Z',
-    geolocation: '43.22,-80.22',
-    duration: 3600,
+    timestamp: '2024-03-26T12:00:000Z',
+    duration: 3600, // 1 hour
+    geolocation: '36.7783,-119.417931', // San Francisco, CA
   },
-]);
+  // Add more input parameters as needed
+];
+const result = await plugin.execute(inputs);
+
+console.log(result);
 ```
 
 ### Manifest Usage
 
-#### Input for manifest
+#### Input
 
 ```yaml
-inputs:
-  - timestamp: 2021-01-01T00:00:00Z
-    geolocation: '43.22,-80.22'
-    duration: 3600
+name: watt-time
+description: simple demo invoking watt-time
+tags:
+initialize:
+  plugins:
+    watt-time:
+      method: WattTimeGridEmissions
+      path: '@grnsft/if-unofficial-plugins'
+  outputs:
+    - yaml
+tree:
+  children:
+    child:
+      pipeline:
+        - watt-time
+      inputs:
+        - timestamp: '2024-03-05T00:00:00.000Z'
+          duration: 3600
+          geolocation: 36.7783,-119.417931
 ```
 
-## Example manifest
+#### Output
 
 ```yaml
 name: watt-time
 description: simple demo invoking watt-time
-tags:
+tags: null
 initialize:
   plugins:
     watt-time:
-      method: WattTimeGridEmissions
       path: '@grnsft/if-unofficial-plugins'
+      method: WattTimeGridEmissions
+  outputs:
+    - yaml
+if-version: v0.3.1
```

Comment:
do we need to include version in output?

Reply:
it is the actual yaml output and there the `if-version` persists

Reply:
yeah but you have to every time update the version after releases, so it's better to exclude it